client/src/elm/Backend/Update.elm

```diff
@@ -4271,15 +4280,49 @@ generatePrenatalAssessmentMsgs currentDate language isChw activePage updateAsses
                                             |> EverySet.fromList
+
+                                    diabetesDiagnoses =
+                                        EverySet.toList reportedDiagnoses
+                                            |> List.any
+                                                (\diagnosis ->
+                                                    List.member diagnosis Pages.Prenatal.Utils.diabetesDiagnoses
+                                                )
+
+                                    rhNegativeDiagnosis =
+                                        EverySet.member DiagnosisRhesusNegative reportedDiagnoses
                                 in
-                                if not <| EverySet.isEmpty reportedDiagnoses then
-                                    [ Backend.PrenatalEncounter.Model.SetPastPrenatalDiagnoses reportedDiagnoses
-                                        |> Backend.Model.MsgPrenatalEncounter originatingEncounterId
-                                        |> App.Model.MsgIndexedDb
-                                    ]
+                                if EverySet.isEmpty reportedDiagnoses then
+                                    []
 
-                                else
+                                else if
+                                    (-- Reporting back about previous diagnois results in hospital referral
+                                     -- at Next steps. For Diabetes, we have logic saying that when it's first
+                                     -- diagnosed, patient is referred to hospital.
+                                     -- On next occasions, no next steps are required.
+                                     -- Therefore, if we know that Diabetes was already diagnosed, we will not
+                                     -- report back about this diagnosis, to prevent unnecessary referral to hospital.
+                                     diabetesDiagnoses
+                                        && Pages.Prenatal.Utils.diagnosedPreviouslyAnyOf Pages.Prenatal.Utils.diabetesDiagnoses assembledAfter
+                                    )
+                                        || (-- Reporting back about previous diagnois results in hospital referral
+                                            -- at Next steps.
+                                            -- Even though Blood group and rhesus test is supposed to run once,
+                                            -- we continue offering it until we get a result.
+                                            -- This way, theoretically, it's possible to have multiple tests pending,
+                                            -- and results can be entered multiple times.
+                                            -- Therefore, if we know that Rhesus Negative was already diagnosed, we will not
+                                            -- report back about this diagnosis, to prevent unnecessary referral to hospital.
```

Comment:
```suggestion
                                        || (-- Reporting back about previous diagnosis results in hospital referral
                                            -- at Next steps.
                                            -- Even though the Blood group and rhesus test are supposed to be run once,
                                            -- we continue offering it until we get a result.
                                            -- This way, theoretically, it's possible to have multiple tests pending,
                                            -- and results can be entered multiple times.
                                            -- Therefore, if we know that Rhesus Negative was already diagnosed, we will not
                                            -- report back about this diagnosis, to prevent unnecessary referral to the hospital.
```